# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: tools/profiler/core/ProfileBuffer.cpp
# Commit: 366030bd2d84
# Full Hash: 366030bd2d84770b99391ab5b3dee5b658ba9624
# Author: Gerald Squelart <gsquelart@mozilla.com>
# Date: 2019-09-17 15:54:53
# Regressor Bug: 1576551
# File Overlap Count: 1
# Description:
#   Bug 1576551 - Assume capacity is in 8-byte entries - r=gregtatum
#   
#   Now that we are using a byte-oriented `BlocksRingBuffer` instead of an array of
#   9-byte `ProfileBufferEntry`'s, internally the profiler sets the buffer size in
#   bytes. However all external users (popup, tests, etc.) still assume that the
# ==============================================================================

diff -r d7ee4148aad9 -r 366030bd2d84 tools/profiler/core/ProfileBuffer.cpp
--- a/tools/profiler/core/ProfileBuffer.cpp	Tue Sep 17 01:49:59 2019 +0000
+++ b/tools/profiler/core/ProfileBuffer.cpp	Tue Sep 17 01:50:19 2019 +0000
@@ -165,10 +165,15 @@
 }
 
 ProfilerBufferInfo ProfileBuffer::GetProfilerBufferInfo() const {
-  return {
-      BufferRangeStart(), BufferRangeEnd(), mEntries.BufferLength()->Value(),
-      mIntervalsNs,       mOverheadsNs,     mLockingsNs,
-      mCleaningsNs,       mCountersNs,      mThreadsNs};
+  return {BufferRangeStart(),
+          BufferRangeEnd(),
+          mEntries.BufferLength()->Value() / 8,  // 8 bytes per entry.
+          mIntervalsNs,
+          mOverheadsNs,
+          mLockingsNs,
+          mCleaningsNs,
+          mCountersNs,
+          mThreadsNs};
 }
 
 /* ProfileBufferCollector */